README.md

```diff
@@ -51,41 +49,158 @@ For .NET Core CLI, use the following command:
 
-- **Production**: The production environment are used for live applications and real-world scenarios. It is recommended to use this environment for production-ready applications. The frontend URL for the production environment is [https://app.tagshelf.com](https://app.tagshelf.com)</br></br>
+To initialize the client for OAuth authentication, specify the credentials explicitly:
+
+```csharp
+var alfred = await AlfredFactory.CreateWithOAuthAsync("username", "password", EnvironmentType.Production);
+```
 
-- **Staging**: The staging environment is used for testing and development purposes. It is recommended to use this environment for testing and development scenarios. The frontend URL for the staging environment is [https://staging.tagshelf.com](https://staging.tagshelf.com)</br></br>
+|HMAC||
+|--|--|
+| **When to Use**    | Best for applications needing individual account access without token management, ensuring secure transactions.|
+| **Benefits**       | Secure request authentication with a unique signature per request.|
+| **Best Practices** | Secure key management, include timestamps to prevent replay attacks, and use a dynamic nonce for each request.|
 
-### Authentication Methods
+To initialize the client for HMAC authentication, provide the secret key and public key:
 
-To obtain the necessary credentials for the following authentication methods, please refer to the [Alfred API documentation](https://docs.tagshelf.dev/authentication) or contact the Alfred support team.
+```csharp
+var alfred = await AlfredFactory.CreateWithHmacAsync("secret_key", "public_key", EnvironmentType.Production);
+```
 
-The following examples demonstrate how to initialize the Alfred client with different authentication methods:
+|App API Key||
+|--|--|
+| **When to Use**    | Recommended for server-to-server communication, automated processes, and background services where user interaction is not feasible.|
+| **Benefits**       | Non-interactive yet robust authentication, simplicity, and easy access control.|
+| **Best Practices** | Secure storage of API keys, apply the principle of least privilege, and regularly rotate keys to minimize compromise risks.|
 
-For OAuth authentication, specify the method and credentials explicitly and provide the environment type (Production or Staging):
+To initialize the client for API key authentication, provide the API key:
 
 ```csharp
-var alfred = await AlfredFactory.CreateWithOAuthAsync("username", "password", EnvironmentType.Production);
+var alfred = await AlfredFactory.CreateWithApiKeyAsync("api_key", EnvironmentType.Staging);
+```
+
+# Getting Started
+
+This library covers 6 different domains: 
+- [Tagshelf](#tagshelf-tagshelf-domain)
+- [Deferred Sessions](#deferred-sessions-deferred-sessions-domain)
+- [Files](#files-files-domain)
+- [Jobs](#jobs-jobs-domain)
+- [Data Points](#data-points-data-points-domain)
+- [Account](#account-account-domain)
+
+## Tagshelf {#tagshelf-domain}
```

Comment:
@rroa please check out the structure of how I'm documenting each domain. Take this first one, the Tagshelf domain as an example. From this line to `line 165`

1. Brief Description of Domain
2. Different Endpoints
    - Brief one line endpoint description
    - Code block that can be copy pasted as-is and runs without errors.
    - Result block of the code